SortingAlgorithms/mergesort.c

```diff
@@ -0,0 +1,82 @@
+#include <stdio.h>
+#include <math.h>
+
+void merge(int array[], int left, int mid, int right)
+{
+	int leftSize = (mid - left + 1),i;
+	int rightSize = (right - mid);
+
+	int leftArray[leftSize], rightArray[rightSize];
+
+	for (i=0; i<leftSize; i++)
+		leftArray[i] = 0;
+	for (i=0; i<rightSize; i++)
+		rightArray[i] = 0;
+
+	int leftIndex=0, rightIndex=0,index;
+
+	//Creates the left and right subarrays
+	for (index=0; index<leftSize; index++)
+		leftArray[index] = array[left + index];
+
+	for (index=0; index<rightSize; index++)
+		rightArray[index] = array[mid + 1 + index];
+
+	int mainArrayIndex = left;
+
+	//Code for merging the sorted subarrays
+	while(leftIndex<leftSize && rightIndex<rightSize)
+	{
+		if(leftArray[leftIndex]>rightArray[rightIndex])
+			array[mainArrayIndex++] = rightArray[rightIndex++];
+		else
+			array[mainArrayIndex++] = leftArray[leftIndex++];
+	}
+
+	//If rightIndex >= rightSize
+	while(leftIndex<leftSize)
+		array[mainArrayIndex++] = leftArray[leftIndex++];
+
+	// If leftIndex >= leftSize
+	while(rightIndex<rightSize)
+		array[mainArrayIndex++] = rightArray[rightIndex++];
+
+	return;
+}
+
+void callMergeSort(int array[], int left, int right)
+{
+	if(left<right)
+	{
+		int mid = (int)floor((left+right-1)/2);
+		// Splits the array into two and calls mergesort on the left and right halves
+		callMergeSort(array, left, mid);
+		// Splits the array into two and calls mergesort on the left and right halves
+		callMergeSort(array, mid+1, right);
+		//Merges the two sorted arrays
+		merge(array, left, mid, right);
+	}
+	return;
```

Comment:
Same as above

Reply:
The second part is for merging the left and right sorted arrays, and in this merge function, once leftIndex>=leftSize, the contents of rightArray are written into array.

I don't get it, what's the problem?

Reply:
Why do you have a unnecessary return at the end of a _void_ function?

Reply:
Dude, that's my coding style. There's no right or wrong as far as that is concerned. 

Please raise genuine issues only :)

Reply:
@arnavb, can you approve the pull request?

Reply:
@Remorax I cannot approve the pull request; I don't have write access to the repo.

Reply:
It doesn't make a difference whether I have it there or not.

Reply:
If you want it to go away, just add another commit, and mine suggestions will become outdated.

Reply:
I thought the PR can't be merged till all suggested changes have been removed. If it doesn't make a difference, then its' okay I guess :P.